consensus/src/pending_order_votes.rs

```diff
@@ -62,7 +63,7 @@ impl PendingOrderVotes {
     pub fn insert_order_vote(
         &mut self,
         order_vote: &OrderVote,
-        validator_verifier: &ValidatorVerifier,
+        epoch_state: Arc<EpochState>,
```

Comment:
I don't understand why this needs to be Arc instead of reference now

Reply:
The validator verifier contains the list of malicious authors (pessimisitc_verify_set). The aggregate_and_verify function sets this information when it observes malicious authors. The epoch manager uses this information when verifying messages. We need to make sure both of them have the same view of epoch state.

Can this be achieved with just passing a reference instead of Arc? My thought process was that if just a reference is passed and this reference is cloned for some reason, then epoch manager and the aggregate_and_verify function are accessing completely different memory locations.

Reply:
yes we need to guarantee the singleton of the EpochState, the simplest way to guarantee it is to remove the Clone from EpochState, so it can only be shared with Arc and can't clone from a reference.

Reply:
Basically, if we do that, then we need to change the syntax of most functions take Arc<EpochState> as input? So, we will have to modify a lot of Dag code as well. Currently, I tried to minimize those changes by modifying only the paths required for the current consensus protocol.

Reply:
no? existing function can still take Arc, we can still clone Arc, just can't clone the EpochState itself

Reply:
I mean if the "Clone" option is completely removed from the traits supported by EpochState, then even the DAG code which uses EpochState now has to be modified to use Arc<EpochState> everywhere even though the DAG doesn't use optimistic signature verification. I thought minimizing the changes is cleaner. Do you prefer to use Arc<EpochState> everywhere?

Reply:
I think Arc everywhere is cleaner, it's probably a big change, so pull it out as a separate pr is better than embedding it here

Reply:
Yeah, it's actually a big change. I tried it before. After changing a few dozen files, I scrapped it and resorted to minimal changes. I can stack on top of this PR if you prefer to implement it.

Reply:
just send out a separate PR instead of stack, we can land that one first and rebase